gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -260,85 +258,73 @@ impl GatewayLightning for ClnRpcService {
-                // Translate action request into a cln rpc response for `htlc_accepted` event
-                let htlca_res = match action {
-                    Some(Action::Settle(Settle { preimage })) => {
-                        if let Ok(pk) = XOnlyPublicKey::from_slice(&preimage) {
-                            serde_json::json!({ "result": "resolve", "payment_key": pk.to_string() })
-                        } else {
-                            temp_node_failure()
-                        }
-                    }
-                    Some(Action::Cancel(Cancel { reason: _ })) => {
-                        // TODO: Translate the reason into a BOLT 4 failure message
-                        // See: https://github.com/lightning/bolts/blob/master/04-onion-routing.md#failure-messages
-                        temp_node_failure()
-                    }
-                    None => {
-                        error!("No action specified for intercepted htlc id: {:?}", hash);
-                        results.push(Err(Status::internal(
-                            "No action specified on this intercepted htlc",
-                        )));
-                        continue;
-                    }
-                };
+        } = request.into_inner();
 
-                // Send translated response to the HTLC interceptor for submission to the cln rpc
-                match outcome.send(htlca_res) {
-                    Ok(_) => {
-                        results.push(Ok(CompleteHtlcsResponse {}));
-                    }
-                    Err(e) => {
-                        error!(
-                            "Failed to send htlc_accepted response to interceptor: {:?}",
-                            e
-                        );
-                        results.push(Err(Status::internal(
-                            "Failed to send htlc_accepted outcome to interceptor",
-                        )));
-                    }
-                };
-                continue;
+        let hash = match sha256::Hash::from_slice(&intercepted_htlc_id) {
+            Ok(hash) => hash,
+            Err(e) => {
+                error!("Invalid intercepted_htlc_id: {:?}", e);
+                return Err(Status::invalid_argument(e.to_string()));
             }
+        };
+
+        if let Some(outcome) = self.interceptor.outcomes.lock().await.remove(&hash) {
+            // Translate action request into a cln rpc response for `htlc_accepted` event
+            let htlca_res = match action {
+                Some(Action::Settle(Settle { preimage })) => {
+                    let assert_pk: Result<[u8; 32], TryFromSliceError> =
+                        preimage.as_slice().try_into();
+                    if let Ok(pk) = assert_pk {
+                        serde_json::json!({ "result": "resolve", "payment_key": pk.to_hex() })
+                    } else {
+                        htlc_processing_failure()
```

Comment:
Could be an `expect`/`panic` since as @dpc noted on a previous PR e.g. sending a too short/long preimage would be a programmer error from the gateway side.

Follow up: at least log with `error` level.